lib/absinthe/federation/schema/entities_field.ex

```diff
@@ -160,4 +180,60 @@ defmodule Absinthe.Federation.Schema.EntitiesField do
         }
       }
     }
+
+  defp reduce_resolution(%{state: :resolved} = res), do: res.value
+
+  defp reduce_resolution(%{middleware: []} = res), do: res
+
+  defp reduce_resolution(%{middleware: [middleware | remaining_middleware]} = res) do
+    call_middleware(middleware, %{res | middleware: remaining_middleware})
+  end
+
+  defp call_middleware({Absinthe.Middleware.Dataloader, {loader, _fun}}, res) do
```

Comment:
```suggestion
  defp call_middleware({Absinthe.Middleware.Dataloader, {loader, _fun}}, %{arguments: %{representations: args}}) do
```
This could remove the `get_in` and then you may need another pattern match below for when this fails to handle it gracefully

Reply:
We will always get arguments here and the `representations` key should always be present since otherwise the validation of the query should pick it up